### PR TITLE
Jonmv/set faiiling revisions aside

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review and merge.

### CODE REVIEW

**Comment by jonmv** [comment]
Urgh, broke a different test ...

**Comment by jonmv** [comment]
There!

**Comment by jonmv** [comment]
Heh, forgot I made the first commit. The second commit is really more of a "don't restart broken revisions, and add test" ... :>

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/Upgrader.java [hunk 1]
```diff
@@ -78,16 +85,20 @@ private InstanceList instances(VersionStatus versionStatus) {
...
                 cancelUpgradesOf(instances.upgradingTo(version.versionNumber()).not().with(UpgradePolicy.canary),
                                  version.versionNumber() + " is broken");
         }
     }
 
-    private void updateTargets(VersionStatus versionStatus, InstanceList instances, UpgradePolicy policy, OptionalInt targetMajorVersion) {
+    private void updateTargets(VersionStatus versionStatus, DeploymentStatusList deploymentStatuses, UpgradePolicy policy, OptionalInt targetMajorVersion) {
+        InstanceList instances = instances(deploymentStatuses);
         InstanceList remaining = instances.with(policy);
+        Instant failureThreshold = controller().clock().instant().minus(Duration.ofDays(5));
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/Upgrader.java:99]: Consider extracting the timeout as a constant.

**Review by mpolden — APPROVED** [review]
👍

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/Upgrader.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/Upgrader.java:99]: Yes ...

**Comment by jonmv** [comment]
Added another minor improvement to the logic as well, PTAL.

**Review by mpolden — APPROVED** [review]
👍

**Pull request merged**